hail/src/main/scala/is/hail/expr/ir/EmitClassBuilder.scala

```diff
@@ -529,6 +532,17 @@ class EmitClassBuilder[C](
     _backendField
   }
 
+  def pool(): Value[RegionPool] = {
+    if (_poolField == null) {
+      cb.addInterface(typeInfo[FunctionWithPool].iname)
```

Comment:
I think you don't need the complexity of the additional interface here. FunctionWithPartitionRegion should add the pool field, I think.